app/javascript/widget/mixins/nextAvailabilityTime.js

```diff
@@ -0,0 +1,256 @@
+      return zonedDate;
+    },
+    presentHour() {
+      return this.newDateWithTimeZone.getHours();
+    },
+    presentMinute() {
+      return this.newDateWithTimeZone.getMinutes();
+    },
+    currentDay() {
+      const date = this.newDateWithTimeZone;
+      const day = date.getDay();
+      const currentDay = Object.keys(this.dayNames).find(
+        key => this.dayNames[key] === this.dayNames[day]
+      );
+      return Number(currentDay);
+    },
+    currentDayOpenHour() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.open_hour
+        : null;
+    },
+    currentDayOpenMinute() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.open_minutes
+        : null;
+    },
+    currentDayCloseHour() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.close_hour
+        : null;
+    },
+    nextDayOpenHour() {
+      if (this.workingHoursEnabled) {
+        return this.nextDayWorkingHours.open_all_day
+          ? 0
+          : this.nextDayWorkingHours.open_hour;
+      }
+      return null;
+    },
+    nextDayOpenMinute() {
+      return this.workingHoursEnabled
+        ? this.nextDayWorkingHours.open_minutes
+        : null;
+    },
+    dayDiff() {
+      // Here this is used to get the difference between current day and next working day
+      if (this.workingHoursEnabled) {
+        const currentDay = this.currentDay;
+        const nextDay = this.nextDayWorkingHours.day_of_week;
+        const totalDays = 6;
+        return nextDay > currentDay
+          ? nextDay - currentDay
+          : totalDays - currentDay + nextDay;
+      }
+      return null;
+    },
+    dayNameOfNextWorkingDay() {
+      if (this.workingHoursEnabled) {
+        return this.dayNames[this.nextDayWorkingHours.day_of_week];
+      }
+      return null;
+    },
+    hoursAndMinutesBackInOnline() {
+      if (this.presentHour >= this.currentDayCloseHour) {
+        return this.getHoursAndMinutesUntilNextDayOpen(
+          this.nextDayOpenHour,
+          this.nextDayOpenMinute,
+          this.currentDayCloseHour
+        );
+      }
+      return this.getHoursAndMinutesUntilNextDayOpen(
+        this.currentDayOpenHour,
+        this.currentDayOpenMinute,
+        this.currentDayCloseHour
+      );
+    },
+    exactTimeInAmPm() {
+      return `${this.timeSlot.day !== this.currentDay ? 'tomorrow' : ''} ${
+        this.timeSlot.day !== this.currentDay ? '' : `at ${this.timeSlot.from}`
+      }`;
+    },
+    hoursLeftValue() {
+      const {
+        hours: hoursLeft,
+        minutes: minutesLeft,
+      } = this.hoursAndMinutesBackInOnline;
+      const hourString = hoursLeft === 1 ? 'hour' : 'hours';
+      const roundedUpMinLeft = Math.ceil(minutesLeft / 5) * 5; // round up minutes to nearest multiple of 5
+      return `in ${hoursLeft} ${hourString}${
+        minutesLeft > 0 ? ` and ${roundedUpMinLeft} minutes` : ''
+      }`;
```

Comment:
These methods can be collapsed into one. Assuming that the `hours` value from `this. hoursAndMinutesBackInOnline()` is 0

```js
    hoursAndMinutesLeft() {
      const { hours, minutes } = this.hoursAndMinutesBackInOnline;

      const timeLeftChars = ['in'];

      if (hours > 0) {
        const hourString = hours === 1 ? 'hour' : 'hours';
        timeLeftChars.push(`${hours} ${hourString}`);
      }

      if (minutes > 0) {
        const roundedUpMinLeft = Math.ceil(minutes / 5) * 5;
        timeLeftChars.push(`${roundedUpMinLeft} minutes`);
      }

      return timeLeftChars.join(' ');
    },
```

And the `hoursAndMinutesToBack` can look like this

```js
    hoursAndMinutesToBack() {
      const {
        hours: hoursLeft,
        minutes: minutesLeft,
      } = this.hoursAndMinutesBackInOnline;
      if (hoursLeft > 3) {
        return this.exactTimeInAmPm;
      }
      if (hoursLeft > 0 || minutesLeft > 0) {
        return this.hoursAndMinutesLeft;
      }
      return 'in some time';
    },
```